Fix erroneous change of constructor signature.

kahaaga opened the pull request:
https://github.com/JuliaDynamics/ComplexityMeasures.jl/pull/422 changed the following

```Julia
  function Probabilities(x::AbstractArray{T, N},
            outcomes::Tuple{Vararg{V, N} where V},
            dimlabels::NTuple{N, S};
            normed::Bool = false)
```

to 

```Julia
 function Probabilities(x::AbstractArray{T, N},
            outcomes::Tuple{Vararg{V, N}} where V,
            dimlabels::NTuple{N, S};
            normed::Bool = false)
```

The change was `outcomes::Tuple{Vararg{V, N} where V}` => `outcomes::Tuple{Vararg{V, N}} where V`. 

The first part of the PR does address the deprecation warning the PR was supposed to solve (https://github.com/JuliaDynamics/ComplexityMeasures.jl/issues/419). However, the second change (the above), is not related to the deprecation warning. More seriously, this fundamentally changes the meaning of the type annotation, which has functional consequences for code that depends on this constructor: 
- `Tuple{Vararg{V, N} where V}` (with the `where V` inside the outer brackets) allows the element type of the outcome vectors to be variables (e.g. the first outcome vector may be a Vector{Int}, while the second may be Vector{String}).
-   `Tuple{Vararg{V, N}} where V` (with the `where V` outside the outer brackets) enforces identical element types for all outcome vectors.

Upstream in Associations.jl, this means that we can't do categorical probability estimation with input vectors of arbitrary `eltype` any longer. In turn, this breaks many discrete probabilities-based association based measures. 

I'm here reverting the change and merging immediately. 

@max-de-rooij, I'm not sure if you were aware of this distinction, so I'm tagging you here to let you know. And to be clear: you are not to blame for this error. We should have caught it during the PR review :) 

@Datseris  The reason we didn't catch this is that we don't explicitly test anything that uses this distinction in ComplexityMeasures.jl. We implement and test multivariate probabilities estimation in Associations.jl, not here. I first caught this while upgrading the combat for ComplexityMeasures.jl to v3.7 in Associations.jl (where the tests *did* fail).

Datseris commented on CHANGELOG.md:
```diff
@@ -2,6 +2,11 @@
 
 Changelog is kept with respect to version 0.11 of Entropies.jl. From version v2.0 onwards, this package has been renamed to ComplexityMeasures.jl.
 
+## 3.8
+
+- Fixed a bug in the `Probabilities` constructor. It is now again possible to construct `Probabilities`
+  when outcome vectors have different element types.
+ 
```
Bug fixes do not increment minor number, they increment patch number, see https://semver.org/

Datseris replied:
I don't think we should state this in the changelog either. In my opinion changelog is really about the things a user would care to read about, and bugfixes is not it. they inflate the changelog unecessarily.

Datseris reviewed (changes requested):
You need to add a test here that catches the error that would otherwise exist. If the constructor is here, a test of it needs to be here as well.

kahaaga commented:
> You need to add a test here that catches the error that would otherwise exist. If the constructor is here, a test of it needs to be here as well.

I added a test, removed the changelog entry and incremented only patch version.

kahaaga commented:
Will merge when these tests pass.
